diff --git a/arch/arm/boot/dts/allwinner/sun8i-t113s-mangopi-mq-r-t113.dts b/arch/arm/boot/dts/allwinner/sun8i-t113s-mangopi-mq-r-t113.dts
index 8b3a753..565a129 100644
--- a/arch/arm/boot/dts/allwinner/sun8i-t113s-mangopi-mq-r-t113.dts
+++ b/arch/arm/boot/dts/allwinner/sun8i-t113s-mangopi-mq-r-t113.dts
@@ -15,6 +15,19 @@
 	aliases {
 		ethernet0 = &rtl8189ftv;
 	};
+
+	ir-receiver {
+		compatible = "gpio-ir-receiver";
+		gpios = <&pio 1 2 GPIO_ACTIVE_LOW>; 		/* PB2 */
+		linux,autosuspend-period = <125>;
+	};
+
+	proximity {
+		compatible = "devantech,srf04";
+		trig-gpios = <&pio 1 6 GPIO_ACTIVE_HIGH>; 	/* PB6 */
+		echo-gpios = <&pio 1 7 GPIO_ACTIVE_HIGH>; 	/* PB7 */
+	};
+
 };
 
 &cpu0 {
@@ -33,3 +46,13 @@
 		interrupt-names = "host-wake";
 	};
 };
+
+&gpadc {
+	status = "okay";
+	#address-cells = <1>;
+	#size-cells = <0>;
+	channel@0 {
+        reg = <0>;
+    };
+};
+
